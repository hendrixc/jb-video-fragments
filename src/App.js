import React, { Component } from 'react';
import { ThemeProvider } from '@rmwc/theme';
import Home from './components/home/Home';
import colors from './style/colors';
import 'material-icons/iconfont/material-icons.scss';
import GlobalProvider from './context/GlobalProvider';

class App extends Component {
  render() {
    return (
      <ThemeProvider options={{
        primary: colors.primary800,
        secondary: colors.secondary500,
        onPrimary: colors.primary400,
        textPrimaryOnBackground: colors.black,
      }}
      >
        <GlobalProvider>
          <Home />
        </GlobalProvider>
      </ThemeProvider>
    );
  }
}

export default App;
