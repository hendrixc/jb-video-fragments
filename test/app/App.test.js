import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../../src/App';

describe('ReactDOM testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

jest.mock('../../src/components/clips/ClipList', () => 'clipList');
describe('Snapshot Testing', () => {
  it('renders an home component into the App as snapshot', () => {
    const TextInputComponent = renderer.create(<App />).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
  });
});
