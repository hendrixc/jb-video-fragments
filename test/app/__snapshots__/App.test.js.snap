// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an home component into the App as snapshot 1`] = `
<div
  style={
    Object {
      "--mdc-theme-on-primary": "#78909C",
      "--mdc-theme-on-secondary": "rgba(255, 255, 255, 1)",
      "--mdc-theme-primary": "#37474F",
      "--mdc-theme-secondary": "#E91E63",
      "--mdc-theme-text-primary-on-background": "#000000",
    }
  }
>
  <div
    className="home"
  >
    <div
      className="Header"
    >
      <div
        className="AppBar"
      >
        <header
          className="mdc-top-app-bar"
          style={Object {}}
        >
          <div
            className="mdc-top-app-bar__row"
          >
            <section
              className="mdc-top-app-bar__section"
            >
              <i
                className="rmwc-icon material-icons mdc-top-app-bar__navigation-icon"
                data-mdc-ripple-is-unbounded={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                style={Object {}}
              >
                menu
              </i>
              <div
                className="mdc-top-app-bar__title"
              >
                Video Fragments Tube
              </div>
            </section>
            <section
              className="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
            >
              <i
                className="rmwc-icon material-icons mdc-top-app-bar__navigation-icon"
                data-mdc-ripple-is-unbounded={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                style={Object {}}
              >
                favorite
              </i>
            </section>
          </div>
        </header>
      </div>
    </div>
    <div
      className="content"
    >
      <div
        className="menu"
      >
        <div>
          <aside
            className="mdc-drawer"
          >
            <div
              className="mdc-drawer__header"
            >
              <div
                className="mdc-drawer__title"
              >
                Anonymous User!!
              </div>
              <div
                className="mdc-drawer__subtitle"
              >
                Welcome!!
              </div>
            </div>
            <div
              className="mdc-drawer__content"
            >
              <div
                className="mdc-list"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
              >
                <div
                  className="mdc-list-item"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={Object {}}
                  tabIndex={0}
                >
                  <div
                    className="item"
                  >
                    <i
                      className="rmwc-icon material-icons"
                      style={
                        Object {
                          "color": "#F06292",
                          "paddingRight": 10,
                        }
                      }
                    >
                      favorite
                    </i>
                    <h4>
                      Favorites
                    </h4>
                  </div>
                </div>
                <div
                  className="mdc-list-item"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={Object {}}
                  tabIndex={0}
                >
                  <div
                    className="item"
                  >
                    <i
                      className="rmwc-icon material-icons"
                      style={
                        Object {
                          "color": "#F06292",
                          "paddingRight": 10,
                        }
                      }
                    >
                      search
                    </i>
                    <h4>
                      Search
                    </h4>
                  </div>
                </div>
                <div
                  className="mdc-list-item"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={Object {}}
                  tabIndex={0}
                >
                  <div
                    className="item"
                  >
                    <i
                      className="rmwc-icon material-icons"
                      style={
                        Object {
                          "color": "#F06292",
                          "paddingRight": 10,
                        }
                      }
                    >
                      lock
                    </i>
                    <h4>
                      Account
                    </h4>
                  </div>
                </div>
              </div>
            </div>
          </aside>
        </div>
      </div>
      <div
        className="content"
      >
        <div
          className="videoPlayer"
        >
          <h3>
            Full Video
          </h3>
          <video
            controls={true}
            src="/videos/video.mp4#t=0,52"
          >
            <track
              default={true}
              kind="captions"
            />
          </video>
        </div>
      </div>
      <div
        className="panel border-panel"
      >
        <div>
          <div
            className="add-button"
          >
            <button
              className="mdc-button mdc-button--raised"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={Object {}}
            >
              <span
                className="mdc-button__label"
              >
                <i
                  className="rmwc-icon material-icons mdc-button__icon"
                >
                  add
                </i>
                Add
              </span>
            </button>
          </div>
          <div>
            <div
              className="clip"
            >
              <div
                className="mdc-card"
                style={
                  Object {
                    "width": "20rem",
                  }
                }
              >
                <div
                  className="mdc-card__primary-action"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={Object {}}
                >
                  <section
                    className="mdc-card__media"
                    style={
                      Object {
                        "backgroundImage": "url(/images/asche.jpg)",
                      }
                    }
                  >
                    <div>
                      <p
                        className="name ellipsis"
                      >
                        Full Video
                      </p>
                      <div
                        className="start ellipsis"
                      >
                        Start: 0:00
                      </div>
                      <div
                        className="end ellipsis"
                      >
                        End: 0:52
                      </div>
                    </div>
                  </section>
                </div>
                <section
                  className="mdc-card__actions"
                >
                  <div
                    className="mdc-card__action-icons"
                  >
                    <button
                      aria-hidden="true"
                      className="rmwc-icon material-icons mdc-card__action mdc-card__action--icon mdc-card__action mdc-card__action--icon mdc-icon-button"
                      data-mdc-ripple-is-unbounded={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      style={
                        Object {
                          "color": "#1976D2",
                        }
                      }
                      tabIndex="0"
                    >
                      edit
                    </button>
                    <button
                      aria-hidden="true"
                      className="rmwc-icon material-icons mdc-card__action mdc-card__action--icon mdc-card__action mdc-card__action--icon mdc-icon-button"
                      data-mdc-ripple-is-unbounded={true}
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      style={
                        Object {
                          "color": "#F06292",
                        }
                      }
                      tabIndex="0"
                    >
                      remove
                    </button>
                  </div>
                </section>
              </div>
            </div>
            <div
              className="clip"
            >
              <div
                className="mdc-card"
                style={
                  Object {
                    "width": "20rem",
                  }
                }
              >
                <div
                  className="mdc-card__primary-action"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={Object {}}
                >
                  <section
                    className="mdc-card__media"
                    style={
                      Object {
                        "backgroundImage": "url(/images/asche.jpg)",
                      }
                    }
                  >
                    <div>
                      <p
                        className="name ellipsis"
                      >
                        First part
                      </p>
                      <div
                        className="start ellipsis"
                      >
                        Start: 0:00
                      </div>
                      <div
                        className="end ellipsis"
                      >
                        End: 0:10
                      </div>
                    </div>
                  </section>
                </div>
                <section
                  className="mdc-card__actions"
                >
                  <div
                    className="mdc-card__action-icons"
                  >
                    <button
                      aria-hidden="true"
                      className="rmwc-icon material-icons mdc-card__action mdc-card__action--icon mdc-card__action mdc-card__action--icon mdc-icon-button"
                      data-mdc-ripple-is-unbounded={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      style={
                        Object {
                          "color": "#1976D2",
                        }
                      }
                      tabIndex="0"
                    >
                      edit
                    </button>
                    <button
                      aria-hidden="true"
                      className="rmwc-icon material-icons mdc-card__action mdc-card__action--icon mdc-card__action mdc-card__action--icon mdc-icon-button"
                      data-mdc-ripple-is-unbounded={true}
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      style={
                        Object {
                          "color": "#F06292",
                        }
                      }
                      tabIndex="0"
                    >
                      remove
                    </button>
                  </div>
                </section>
              </div>
            </div>
            <div
              className="clip"
            >
              <div
                className="mdc-card"
                style={
                  Object {
                    "width": "20rem",
                  }
                }
              >
                <div
                  className="mdc-card__primary-action"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={Object {}}
                >
                  <section
                    className="mdc-card__media"
                    style={
                      Object {
                        "backgroundImage": "url(/images/asche.jpg)",
                      }
                    }
                  >
                    <div>
                      <p
                        className="name ellipsis"
                      >
                        Second part
                      </p>
                      <div
                        className="start ellipsis"
                      >
                        Start: 0:22
                      </div>
                      <div
                        className="end ellipsis"
                      >
                        End: 0:35
                      </div>
                    </div>
                  </section>
                </div>
                <section
                  className="mdc-card__actions"
                >
                  <div
                    className="mdc-card__action-icons"
                  >
                    <button
                      aria-hidden="true"
                      className="rmwc-icon material-icons mdc-card__action mdc-card__action--icon mdc-card__action mdc-card__action--icon mdc-icon-button"
                      data-mdc-ripple-is-unbounded={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      style={
                        Object {
                          "color": "#1976D2",
                        }
                      }
                      tabIndex="0"
                    >
                      edit
                    </button>
                    <button
                      aria-hidden="true"
                      className="rmwc-icon material-icons mdc-card__action mdc-card__action--icon mdc-card__action mdc-card__action--icon mdc-icon-button"
                      data-mdc-ripple-is-unbounded={true}
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      style={
                        Object {
                          "color": "#F06292",
                        }
                      }
                      tabIndex="0"
                    >
                      remove
                    </button>
                  </div>
                </section>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
